src/nrniv/cxprop.cpp

```diff
@@ -598,6 +598,38 @@ void nrn_update_ion_pointer(Symbol* sion, Datum* dp, int id, int ip) {
     dp[id].pval = pvar + ip;
 }
 
+
+void nrn_poolshrink(int shrink) {
+    if (shrink) {
+        for (int i = 0; i < npools_; ++i) {
+            auto& pdbl = dblpools_[i];
+            auto& pdatum = datumpools_[i];
+            if ((pdbl && pdbl->nget() == 0)) {
+                nrn_delete_prop_pool(i);
+            }
+            if ((pdatum && pdatum->nget() == 0) && pdatum->size() > 10) {
```

Comment:
I'm curious here why we check that `pdatum->size() > 10`?

Reply:
That is an accidental leftover that should be removed. I had the same condition earlier on line 607 and got rid of it.
Originally I was thinking it wasn't worth shrinking an already small pool.